src/Components/Aspire.MongoDB.Driver/AspireMongoDbDriverExtensions.cs

```diff
@@ -0,0 +1,174 @@
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional delegate that can be used for customizing options. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:MongoDB" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="MongoDbSettings.ConnectionString"/> is not provided.</exception>
+    public static void AddMongoDbDataSource(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<MongoDbSettings>? configureSettings = null)
+        => AddMongoDbDataSource(builder, DefaultConfigSectionName, configureSettings, connectionName, serviceKey: null);
+
+    /// <summary>
+    /// Registers <see cref="IMongoClient"/> and <see cref="IMongoDatabase"/> instances for connecting MongoDB database with MongoDB.Driver client.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional delegate that can be used for customizing options. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:MongoDB:{name}" section.</remarks>
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="MongoDbSettings.ConnectionString"/> is not provided.</exception>
+    public static void AddKeyedMongoDbDataSource(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<MongoDbSettings>? configureSettings = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        AddMongoDbDataSource(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddMongoDbDataSource(
+        IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<MongoDbSettings>? configureSettings,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = builder.GetMongoDbSettings(
+            connectionName,
+            configurationSectionName,
+            configureSettings);
+
+        settings.ValidateSettings(connectionName, configurationSectionName);
```

Comment:
We've been delaying this validation until the app is started. That way any exception can be logged with an ILogger. See https://github.com/dotnet/aspire/pull/213.

Reply:
Done, please review